gcloud/resource_manager/client.py

```diff
@@ -0,0 +1,238 @@
+                  all of the results without thinking about pagination.
+                  Each item will be a :class:`.project.Project` object.
+        """
+        params = {}
+
+        if page_size is not None:
+            params['pageSize'] = page_size
+
+        if filter_params is not None:
+            params['filter'] = filter_params
+
+        client = self
+
+        class ProjectIterator(Iterator):
+            """An iterator over a list of Project resources."""
+
+            def get_items_from_response(self, response):
+                """Yield :class:`.resource_manager.project.Project` items
+                from response.
+
+                :type response: dict
+                :param response: The JSON API response for a page of projects.
+                """
+                for resource in response.get('projects', []):
+                    item = Project.from_api_repr(resource, client=client)
+                    yield item
+
+        return ProjectIterator(connection=self.connection, extra_params=params,
+                               path='/projects')
+
+    def project(self, project_id):
+        """Get a Project instance without making an API call.
+
+        Typically used when creating a new project.
+
+        See :func:`gcloud.resource_manager.client.Client.get_project` if you
+        want to load a project and its metadata using an API call.
+
+        Example::
+
+            >>> client = Client()
+            >>> project = client.project('purple-spaceship-123')
+            >>> print project.name
+            None
+            >>> print project.project_id
+            purple-spaceship-123
+
+        :type project_id: str
+        :param project_id: The ID for this project.
+
+        :rtype: :class:`gcloud.resource_manager.project.Project`
+        :returns: A new instance of a :class:`.project.Project` **without**
+                  any metadata loaded.
+        """
+        return Project(project_id=project_id, client=self)
+
+    def get_project(self, project_id):
+        """Get a Project instance and its metadata via an API call.
+
+        Example::
+
+            >>> client = Client()
+            >>> project = client.get_project('purple-spaceship-123')
+            >>> print project.name
+            Purple Spaceship 123
+            >>> print project.project_id
+            purple-spaceship-123
+
+        See :func:`gcloud.resource_manager.client.Client.project` if you
+        want to load a project **without** its metadata (aka, without an
+        API call).
+
+        :type project_id: str
+        :param project_id: The ID for this project.
+
+        :rtype: :class:`gcloud.resource_manager.project.Project`
+        :returns: A new instance of a :class:`.project.Project` with all
+                  its metadata loaded.
+        """
+        try:
+            project = self.project(project_id)
+            project.reload()
+        except NotFound:
```

Comment:
Storage makes the distinction between `get_bucket` and `lookup_bucket`. Lookup catches the 404 while get does not. It's worth being consistent across packages.